ash/src/extensions/ext/drm_format_modifier.rs

```diff
@@ -0,0 +1,83 @@
+use crate::prelude::*;
+use crate::vk;
+use crate::{Device, Instance};
+use std::ffi::CStr;
+use std::mem;
+
+#[derive(Clone)]
+pub struct DrmFormatModifier {
+    handle: vk::Device,
+    fp: vk::ExtImageDrmFormatModifierFn,
+}
+
+impl DrmFormatModifier {
+    pub fn new(instance: &Instance, device: &Device) -> Self {
+        let handle = device.handle();
+        let fp = vk::ExtImageDrmFormatModifierFn::load(|name| unsafe {
+            mem::transmute(instance.get_device_proc_addr(handle, name.as_ptr()))
+        });
+        Self { handle, fp }
+    }
+
+    /// <https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/vkGetImageDrmFormatModifierPropertiesEXT.html>
+    pub unsafe fn get_image_drm_format_modifier_properties(
+        &self,
+        image: &vk::Image,
+        properties: &mut vk::ImageDrmFormatModifierPropertiesEXT,
+    ) -> VkResult<()> {
+        (self.fp.get_image_drm_format_modifier_properties_ext)(self.handle, *image, properties)
+            .result()
+    }
+
+    pub unsafe fn get_drm_format_properties_list(
+        instance: &Instance,
+        pdevice: vk::PhysicalDevice,
+        format: vk::Format,
+        format_properties: vk::FormatProperties,
+    ) -> Vec<vk::DrmFormatModifierPropertiesEXT> {
+        let mut list = vk::DrmFormatModifierPropertiesListEXT::default();
+
+        // Get count of DRM format modifiers for the format.
+        {
+            let mut format_properties_2 = vk::FormatProperties2::builder()
+                .format_properties(format_properties)
+                .push_next(&mut list);
+
+            instance.get_physical_device_format_properties2(
```

Comment:
This extension requires Vulkan 1.0 yet this function was only stabilized in 1.1. How are you going to communicate this requirement to the caller, or otherwise fall back to the extension (presuming the caller enabled it in the first place...) if the device supports only Vulkan 1.0?

FWIW this function needs a doc-comment anyway.

Reply:
The one solution I could think of for Vulkan 1.0 is essentially duplicating this function inside an `impl GetPhysicalDeviceProperties2 { ... }`